server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -404,6 +415,23 @@ private void failAllocationOfNewPrimaries(RoutingAllocation allocation) {
         }
     }
 
+    private void scheduleRerouteIfAllocatorTimedOut() {
+        if (allocatorTimedOut()) {
+            if (rerouteService == null) {
+                logger.info("RerouteService not set to schedule reroute after allocator time out");
```

Comment:
What does this log denote? I would rather be more assertive if this was expected to be null or not?

Reply:
Updated. Added initially for tests